plugins/mac-videotoolbox/encoder.c

```diff
@@ -1005,6 +1233,107 @@ static obs_properties_t *vt_properties(void *unused, void *data)
 	return props;
 }
 
+#if ENABLE_HEVC
+static obs_properties_t *vt_properties_hevc(void *unused, void *data)
+{
+	UNUSED_PARAMETER(unused);
+	struct vt_encoder_type_data *type_data = data;
+
+	obs_properties_t *props = obs_properties_create();
+	obs_property_t *p;
+
+	vt_properties_bitrate(props, data);
+
+	p = obs_properties_add_list(props, "profile", TEXT_PROFILE,
+				    OBS_COMBO_TYPE_LIST,
+				    OBS_COMBO_FORMAT_STRING);
+	obs_property_list_add_string(p, "main", "main");
+	obs_property_list_add_string(p, "main10", "main10");
+	if (__builtin_available(macOS 12.3, *)) {
+		obs_property_list_add_string(p, "main 4:2:2 10", "main42210");
+	}
+
+	obs_properties_add_bool(props, "bframes", TEXT_BFRAMES);
+
+	return props;
+}
+#endif
+
+static obs_properties_t *vt_properties_prores(void *unused, void *data)
+{
+	UNUSED_PARAMETER(unused);
+	struct vt_encoder_type_data *type_data = data;
+
+	obs_properties_t *props = obs_properties_create();
+	obs_property_t *p;
+
+	p = obs_properties_add_list(props, "codec_type", TEXT_PRORES_CODEC,
+				    OBS_COMBO_TYPE_LIST, OBS_COMBO_FORMAT_INT);
+
+	uint32_t codec_availability_flags = 0;
+	CFArrayRef encoder_list;
+	VTCopyVideoEncoderList(NULL, &encoder_list);
+	CFIndex size = CFArrayGetCount(encoder_list);
+
+	// Build list of types based on availability.
```

Comment:
Could this list be built in obs_module_load and added to the type data or something?
Creating it each time the properties are called sounds quite expensive.